tools/ddspreviewer/ddsheaderparser.cpp

```diff
@@ -0,0 +1,224 @@
+{
+    const int enumIdx = DDSHeaderParser::staticMetaObject.indexOfEnumerator(enumName);
+    const QMetaEnum metaEnum = DDSHeaderParser::staticMetaObject.enumerator(enumIdx);
+    const int keysCount = metaEnum.keyCount();
+
+    for (int i = 0; i < keysCount; ++i) {
+        const uint enumValue = uint(metaEnum.value(i));
+        if (v == enumValue)
+            return QStringLiteral("%1 (%2)").arg(metaEnum.key(i)).arg(v);
+    }
+    return QString::number(v);
+}
+
+QString flagsToEnumName(uint v, const char *enumName)
+{
+    QString str;
+
+    const int enumIdx = DDSHeaderParser::staticMetaObject.indexOfEnumerator(enumName);
+    const QMetaEnum metaEnum = DDSHeaderParser::staticMetaObject.enumerator(enumIdx);
+    const int keysCount = metaEnum.keyCount();
+
+    for (int i = 0; i < keysCount; ++i) {
+        const uint enumValue = uint(metaEnum.value(i));
+        if ((v & enumValue) == enumValue) {
+            if (!str.isEmpty())
+                str += QStringLiteral(", ");
+            str += QStringLiteral("%1 (0x%2)").arg(metaEnum.key(i)).arg(enumValue, 0, 16);
+        }
+    }
+
+    if (!str.isEmpty())
+        return QStringLiteral("%1 - %2").arg(v, 0, 2).arg(str);
+    return QStringLiteral("%1").arg(v, 0, 2);
+}
+
+} // namespace
+
+DDSHeaderParser::DDSHeaderParser(QObject *parent)
+    : QObject(parent)
+{
+    QObject::connect(this, &DDSHeaderParser::filePathChanged,
+                     this, &DDSHeaderParser::parseTextureHeader);
+}
+
+DDSHeaderParser::~DDSHeaderParser()
+{
+}
+
+QUrl DDSHeaderParser::filePath() const
+{
+    return m_filePath;
+}
+
+QString DDSHeaderParser::ddsHeader() const
+{
+    return m_ddsHeader;
+}
+
+void DDSHeaderParser::setFilePath(const QUrl &filePath)
+{
+    if (m_filePath == filePath)
+        return;
+
+    m_filePath = filePath;
+    emit filePathChanged(m_filePath);
+}
+
+void DDSHeaderParser::parseTextureHeader()
+{
+    m_ddsHeader.clear();
+    emit ddsHeaderChanged();
+
+    // 1) Check path and that file exists
+    const QString path = m_filePath.toLocalFile();
+    QFileInfo fInfo(path);
+    // 2) Check extension is .dds
+    if (!fInfo.exists() || fInfo.suffix() != QStringLiteral("dds")) {
+        qWarning() << path << "doesn't exist or is not a DDS file";
+        return;
+    }
+
+    QFile source(path);
+    if (!source.open(QIODevice::ReadOnly)) {
+        qWarning() << "Unable to open DDS File";
+        return;
+    }
+
+    // 3) Parse header
+    DdsHeader header;
+    if ((source.read(reinterpret_cast<char *>(&header), sizeof header) != sizeof header) || (qstrncmp(header.magic, "DDS ", 4) != 0)) {
+        qWarning() << "Invalid DDS Header";
+        return;
+    }
+
+    const quint32 pixelFlags = qFromLittleEndian(header.pixelFormat.flags);
+    const quint32 fourCC = qFromLittleEndian(header.pixelFormat.fourCC);
+
+    QString dx10HeaderStr = QStringLiteral("{}");
+
+    if (fourCC == DdsFourCC('D', 'X', '1', '0')) {
```

Comment:
To be 100% sure this is computed by the compiler only once you probably want a constexpr intermediate:
constexpr DdsFourCC dx10FourCC{'D', 'X', '1', '0'};
if (fourCC == dx10FourCC) {

Maybe with C++20 we won't need this dance anymore...